### PR TITLE
Fix workflow/stage unpublish/trash/deletion prevention

### PR DESCRIPTION
**Pull request by bembelimen**
### Summary of Changes
When trying to unpublish/trash/delete a workflow/stage currently the joomla plugins has a check implemented to prevent the status change if there are items (articles) assigned.

This checks are broken and should be fixed with this PR

### Testing Instructions
Build different scenarios like 2 workflows one assigned to articles, one not. Try to unpublish/trash both workflows. If successful switch places (first workflow has no items assigned).

Same for stages.

Also try to delete assigned workflows.

### Actual result BEFORE applying this Pull Request
Sometimes you can delete workflow/stages with items assigned. Sometimes you can't trash workflows although no items are assigned.

### Expected result AFTER applying this Pull Request
Not possible anymore to trash workflows/stages when articles are assigned.

### CODE REVIEW

**Comment by karo3** [comment]
I have tested this item :red_circle: unsuccessfully on 753a0a6132dc8ecbc5487a17a3b04875d911c412<br /><br />Inconsistent behavior:

If you mark a stage/workflow that is not in use and not default and one that is default and want to delete both, he deletes one stage
If you mark a stage/workflow that is not in use and not default and one that is in use and want to delete both, it does not delete any stage
If you mark a stage/workflow that is not in use and not default and one that is default AND is in use and wants to delete both, he deletes one stage<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31120">issues.joomla.org/tracker/joomla-cms/31120</a>.</sub>

**Comment by HLeithner** [comment]
This pull request has automatically rebased to 4.2-dev.

**Comment by rdeutz** [comment]
@bembelimen should we close this because of the great success with this change :-)

**Comment by bembelimen** [comment]
> I have tested this item red_circle unsuccessfully on [753a0a6](https://github.com/joomla/joomla-cms/commit/753a0a6132dc8ecbc5487a17a3b04875d911c412)Inconsistent behavior:
> 
> If you mark a stage/workflow that is not in use and not default and one that is default and want to delete both, he deletes one stage If you mark a stage/workflow that is not in use and not default and one that is in use and want to delete both, it does not delete any stage If you mark a stage/workflow that is not in use and not default and one that is default AND is in use and wants to delete both, he deletes one stage
> This comment was created with the [J!Tracker Application](https://github.com/joomla/jissues) at [issues.joomla.org/tracker/joomla-cms/31120](https://issues.joomla.org/tracker/joomla-cms/31120).

So updated the PR, it should only unpublish/trash/delete anything of all of the selected items have no assignments.

Please test again.

File: administrator/language/en-GB/com_workflow.ini [hunk 1]
```diff
@@ -46,7 +46,7 @@ COM_WORKFLOW_MODIFIED_AT_ASC="Date Modified ascending"
 COM_WORKFLOW_MODIFIED_AT_DESC="Date Modified descending"
 COM_WORKFLOW_MODIFIED_LABEL="Date Modified"
 COM_WORKFLOW_MSG_DELETE_IS_DEFAULT="You can't delete the default item."
-COM_WORKFLOW_MSG_DELETE_STAGE_IS_ASSIGNED="This stage is in use by the component."
+COM_WORKFLOW_MSG_DELETE_STAGE_IS_ASSIGNED="This stage is assigned to an item of this component and has to stay on a publish state."
```
Comment (chmst) [inline administrator/language/en-GB/com_workflow.ini]: Several states can be selected - maybe better "Selected states are assigned .. and have ... "

**Comment by chmst** [comment]
Line 559 

`                if (!$this->_canDeleteStage($pk)) {`

has to be changed with _workflowNotUsed

File: administrator/language/en-GB/com_workflow.ini (hunk 1 above)
Comment (brianteeman) [reply administrator/language/en-GB/com_workflow.ini]: agree with @chmst and the current proposed string is not correct english
example 
> This stage is assigned to an item of this component and has to stay on a publish state.

==>

> This stage is assigned to an item of this component and has to remain in a publish state.

Comment (chmst) [inline administrator/language/en-GB/com_workflow.ini]:
```suggestion
COM_WORKFLOW_MSG_DELETE_STAGE_IS_ASSIGNED="This stage is assigned to an item of this component and has to remain in a publish state."
```

**Comment by bembelimen** [comment]
Thx

**Comment by drmenzelit** [comment]
I have tested this item :white_check_mark: successfully on ecc347894019f79c8fef06f85fbe4dfda5c3bf4c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31120">issues.joomla.org/tracker/joomla-cms/31120</a>.</sub>

**Comment by chmst** [comment]
I have tested this item :white_check_mark: successfully on ecc347894019f79c8fef06f85fbe4dfda5c3bf4c<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31120">issues.joomla.org/tracker/joomla-cms/31120</a>.</sub>

**Comment by brianteeman** [comment]
My bad. It should be "in a published"

**Comment by chmst** [comment]
RTC<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/31120">issues.joomla.org/tracker/joomla-cms/31120</a>.</sub>

**Pull request merged**